src/main/java/io/appium/java_client/AppiumDriver.java

```diff
@@ -270,4 +277,48 @@ public Response execute(String driverCommand, Map<String, ?> parameters) {
     public Response execute(String command) {
         return super.execute(command, Collections.emptyMap());
     }
+
+    private static File saveToFile(byte[] data) {
+        try {
+            Path tmpFilePath = Files.createTempFile("screenshot", ".png");
+            File tmpFile = tmpFilePath.toFile();
+            tmpFile.deleteOnExit();
+            Files.write(tmpFilePath, data);
+            return tmpFile;
+        } catch (IOException e) {
+            throw new WebDriverException(e);
+        }
+    }
+
+    @Override
+    public <X> X getScreenshotAs(OutputType<X> outputType) {
+        Response response = execute(DriverCommand.SCREENSHOT);
+        Object result = response.getValue();
+        final byte[] screenshotBase64Bytes;
+        if (result instanceof String) {
+            screenshotBase64Bytes = ((String) result).getBytes(StandardCharsets.UTF_8);
+        } else if (result instanceof byte[]) {
+            screenshotBase64Bytes = (byte[]) result;
+        } else {
+            throw new RuntimeException(
+                    String.format("Unexpected result for %s command: %s", DriverCommand.SCREENSHOT,
+                        result == null ? "null" : (result.getClass().getName() + " instance"))
+            );
+        }
+        // TODO: Eventually we should not override this method.
+        // TODO: Although, we have a legacy burden,
+        // TODO: so it's impossible to do it the other way as of Oct 29 2022.
+        // TODO: See https://github.com/SeleniumHQ/selenium/issues/11168
+        byte[] screenshotRawBytes = Base64.getMimeDecoder().decode(screenshotBase64Bytes);
+        if (outputType == OutputType.BYTES) {
+            return (X) screenshotRawBytes;
+        } else if (outputType == OutputType.BASE64) {
+            //noinspection unchecked
+            return (X) Base64.getEncoder().encodeToString(screenshotRawBytes);
+        } else if (outputType == OutputType.FILE) {
+            //noinspection unchecked
+            return (X) saveToFile(screenshotRawBytes);
+        }
+        throw new IllegalArgumentException(String.format("Unsupported %s instance", OutputType.class.getName()));
```

Comment:
what about making it simpler?:

```suggestion
        return super.getScreenshotAs(new OutputType<>() {
            @Override
            public X convertFromBase64Png(String base64Png) {
            // TODO: Eventually we should not override this method.
            // TODO: Although, we have a legacy burden,
            // TODO: so it's impossible to do it the other way as of Oct 29 2022.
            // TODO: See https://github.com/SeleniumHQ/selenium/issues/11168
                return outputType.convertFromBase64Png(base64Png.replaceAll("\r?\n", ""));
            }

            @Override
            public X convertFromPngBytes(byte[] png) {
                return outputType.convertFromPngBytes(png);
            }
        });
```

Reply:
good idea. Changed